client/packages/common/src/plugins/pluginProvider.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  create,
+  FrontendPluginMetadataNode,
+  isArray,
+  mergeWith,
+  Plugins,
+} from '@openmsupply-client/common';
+import { Environment } from 'packages/config/src';
+
+// PLUGIN PROVIDER
+type PluginProvider = {
+  plugins: Plugins;
+  addPlugins: (_: Plugins) => void;
+};
+
+export const usePluginProvider = create<PluginProvider>(set => {
+  return {
+    plugins: {},
+    addPlugins: plugins => {
+      set(state => {
+        // Here can determine if version is suitable
+        const newPlugins = mergeWith(state.plugins, plugins, (a, b) =>
+          isArray(a) ? a.concat(b) : undefined
+        );
+
+        return { ...state, plugins: newPlugins };
+      });
+    },
+  };
+});
+
+// PLUGINS INIT
+
+// LOAD REMOTE PLUGIN
```

Comment:
This is functionality similar to what was there before, most of the code is taking from federation module examples, and based on: https://github.com/andreievg/plugins-example

There is no lazy loading anymore, which simplifies things a bit